test_fms/horiz_interp/test_horiz_interp.F90

```diff
@@ -1100,21 +1101,25 @@ subroutine test_assignment()
         call horiz_interp_del(Interp_cp)
         ! test deletion after direct calls
         ! this set up is usually done within horiz_interp_new
-        nlon_in  = size(lon_in_1d(:))-1;  nlat_in  = size(lat_in_1d(:))-1
-        nlon_out = size(lon_out_1d(:))-1; nlat_out = size(lat_out_1d(:))-1
+        nlon_in  = size(lon_in_1d(:));  nlat_in  = size(lat_in_1d(:))
+        nlon_out = size(lon_out_1d(:)); nlat_out = size(lat_out_1d(:))
         allocate(lon_src_1d(nlon_in), lat_src_1d(nlat_in))
         allocate(lon_dst_1d(nlon_out), lat_dst_1d(nlat_out))
-        do i = 1, nlon_in
+        do i = 1, nlon_in-1
           lon_src_1d(i) = (lon_in_1d(i) + lon_in_1d(i+1)) * 0.5_lkind
```

Comment:
If `nlon_in` is equal to the size of `lon_in_1d` and this this is looping from `1 to nlon_in`, wouldn't this cause issues when `i = nlon_in` as `lon_in_1d(i+1)` does not exist?

Reply:
Do you mean the loop on line 1108? It's going from `1` to `nlon_in-1`. It should only access up to `lon_in_1d(nlon_in)` on the last iteration.

Reply:
lol i read this completely wrong. sorry about that.